maven-jxr-plugin/src/test/java/org/apache/maven/plugin/jxr/stubs/AggregateTestMavenProjectStub.java

```diff
@@ -62,15 +62,15 @@ public AggregateTestMavenProjectStub()
         setInceptionYear( model.getInceptionYear() );
 
         String basedir = getBasedir().getAbsolutePath();
-        List compileSourceRoots = new ArrayList();
+        List<String> compileSourceRoots = new ArrayList<>();
         compileSourceRoots.add( basedir + "/src/test/resources/unit/aggregate-test/aggregate/test" );
         setCompileSourceRoots( compileSourceRoots );
 
         //set the report plugins
-        List reportPlugins = new ArrayList();
-        for ( Iterator iter = model.getReporting().getPlugins().iterator(); iter.hasNext(); )
+        List<ReportPlugin> reportPlugins = new ArrayList<>();
+        for ( Iterator<ReportPlugin> iter = model.getReporting().getPlugins().iterator(); iter.hasNext(); )
```

Comment:
Why not use foreach directly when using generics?

Reply:
Mostly I was just fixing the warnings, but it turns out we can go even further and remove the loop completely, just use the constructor. Done.